test/ecto/query/planner_test.exs

```diff
@@ -1275,4 +1275,53 @@ defmodule Ecto.Query.PlannerTest do
       from(p in Post, order_by: p.title) |> normalize(:delete_all)
     end
   end
+
+  describe "normalize: subqueries in boolean expressions" do
+    test "replaces {:subquery, index} with an Ecto.SubQuery struct" do
+      subquery = from(p in Post, select: p.visits)
+
+      %{wheres: [where]} =
+        from(p in Post, where: p.visits >= all(subquery))
+        |> normalize()
+
+      case where.expr do
+        {:>=, _, [_, {:all, _, [%Ecto.SubQuery{}] }]} ->
+          :ok
+
+        _ ->
+          raise "planner did not replace {:subquery, index} with an Ecto.SubQuery struct in quantified comparison"
+      end
+
+      %{wheres: [where]} =
+        from(p in Post, where: exists(subquery))
+        |> normalize()
+
+      case where.expr do
+        {:exists, _, [%Ecto.SubQuery{}]} ->
+          :ok
+
+        _ ->
+          raise "planner did not replace {:subquery, index} with an Ecto.SubQuery struct in exists clause"
+      end
+    end
+
+    test "raises a runtime error if more than 1 field is selected" do
+      s = from(p in Post, select: [p.visits, p.id])
+
+      assert_raise Ecto.QueryError, fn ->
+        from(p in Post, where: p.id in subquery(s))
+        |> normalize()
+      end
+
+      assert_raise Ecto.QueryError, fn ->
+        from(p in Post, where: p.id > subquery(s))
```

Comment:
```suggestion
        from(p in Post, where: p.id > any(s))
```

?

Reply:
This is valid SQL assuming the subquery returns a single column and a single row. If the database finds that the subquery returns more than 1 row it'll throw a cardinality exception.

Reply:
Right, in this case I would rather have folks using all/any with proper limit, rather than support a syntax designed for row constructs which we don't properly handle. :D